### PR TITLE
host - improve validation and conversion of MAC addresses

### PR DESCRIPTION
**Pull request by evgeni**
this is currently blocked on https://github.com/ansible/ansible/issues/72642 as `validate-modules` seems to barf on callable validators

### CODE REVIEW

**Comment by evgeni** [comment]
actually blocked on https://github.com/ansible/ansible/issues/66968, the first one was a dupe, sorry

File: plugins/modules/host.py [hunk 1]
```diff
@@ -217,7 +237,7 @@ def main():
             managed=dict(type='bool'),
             build=dict(type='bool'),
             ip=dict(),
-            mac=dict(),
+            mac=dict(type=mac),
```
Comment (mdellweg) [inline plugins/modules/host.py:467]: So this is how you add your own validation to ansible parameters?
You really always keep learning. ;)

Comment (evgeni) [reply plugins/modules/host.py:467]: Yes, until it fucks up `validate-modules` :(

**Comment by evgeni** [comment]
I'll try to persuade ansible to fix that properly.

**Comment by evgeni** [comment]
https://github.com/ansible/ansible/issues/66968 has been closed, so :woman_shrugging:

**Pull request closed without merging**